docs/operator-manual/declarative-setup.md

```diff
@@ -77,7 +77,7 @@ spec:
 ```
 
 !!! warning
-    By default, deleting an application will not perform a cascade delete, which would delete its resources. You must add the finalizer if you want this behaviour - which you may well not want.
+    By default, deleting an application will not perform a cascade delete, which would delete its resources. If you prefer a cascade delete of its resource, you must add the finalizer:
```

Comment:
I wonder if it makes sense to instead link out to the App deletion section for this.

Also based on https://github.com/argoproj/argo-cd/pull/10949, it may be good to move away from "by default" and mention specifically if the finalizer is present.

```suggestion
    Without the `resources-finalizer.argocd.argoproj.io` finalizer, deleting an application will not delete the resources it manages. To perform a cascading delete, you must add the finalizer. See [App Deletion](../user-guide/app_deletion.md#about-the-deletion-finalizer).
```